syntax = "proto3";

// import "job.proto";

option java_multiple_files = true;
option java_package = "de.tum.in.ase.athene.protobuf";

message Segment {
    string id = 1;
    optional int32 submissionId = 2;
    optional string text = 3;
    optional int32 startIndex = 4;
    optional int32 endIndex = 5;
}

message DistanceMatrixEntry {
    int32 x = 1;
    int32 y = 2;
    float value = 3;
}

message Cluster {
    int32 treeId = 1;
    repeated Segment segments = 2;
    repeated DistanceMatrixEntry distanceMatrix = 3;
}

message ClusterNode {
    int32 parent = 1;
    int32 child = 2;
    float lambdaVal = 3;
    int32 childSize = 4;
}

/*
message ClusteringRequest {}
message ClusteringResponse {
    Job job = 1;
    repeated ClusterNode clusterTree = 2;
    repeated DistanceMatrixEntry distanceMatrix = 3;
    repeated Cluster clusters = 4;
}

service ClusteringService {
    rpc cluster(ClusteringRequest) returns (ClusteringResponse) {}
}
*/

message AtheneResponse {
    repeated Segment segments = 1;
    repeated Cluster clusters = 2;
    repeated DistanceMatrixEntry distanceMatrix = 3;
    repeated ClusterNode clusterTree = 4;
}